### ISSUE TITLE
[ARM breakage] Same old same old 

### ISSUE THREAD

**Issue by petevine** [gfx-rs/gfx#844]
More fallout from https://github.com/rust-lang/rust/issues/29867

```` 
Compiling gfx_device_gl v0.7.0
  /home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:314:21: 314:69 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:314                     target.as_bytes_with_nul().as_ptr() as *const i8);
                                                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:314:21: 314:69 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:328:46: 328:94 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:328                 gl.GetFragDataLocation(name, target.as_bytes_with_nul().as_ptr() as *const i8)
                                                                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:328:46: 328:94 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/info.rs:145:55: 145:60 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/info.rs:145     mem::transmute(str::from_utf8(ffi::CStr::from_ptr(c_str).to_bytes()).unwrap())
                                                                                                                                                       ^~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/info.rs:145:55: 145:60 help: run `rustc --explain E0308` to see a detailed explanation
````

Using nightly rust 1.7 on armv7 Linux.

**Comment by kvark** [comment]
Thanks for the report!

**Comment by petevine** [comment]
Hi,

A quick and sufficient fix is to cast every instance to `as *const _` whereas if you wanted to get rid of the problem altogether here's the latest [solution](https://github.com/vberger/wayland-client-rs/issues/42).

**Comment by kvark** [comment]
@petevine thanks! please consider filing a PR

**Comment by petevine** [comment]
No prob! You're probably aware of the [issue](https://github.com/rust-lang/rust/issues/30713) which led to the travis [failure](https://travis-ci.org/gfx-rs/gfx/builds/100964922). On my local `1.7` from before that change all compiles cleanly.

**Comment by andresv** [comment]
It fails also with rust-1.6-stable on RPi2:

```
   Compiling gfx_device_gl v0.7.0
/home/pi/.multirust/toolchains/1.6/cargo/registry/src/github.com-48ad6e4054423464/gfx_device_gl-0.7.0/src/shade.rs:314:21: 314:69 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
```

**Comment by petevine** [comment]
I didn't follow the situation after the PR that'd fixed this issue got reverted.

@andresv Do you mean there's just one error now or you simply pasted a snippet?

**Comment by andresv** [comment]
I just pasted the first error.

**Comment by petevine** [comment]
Nope, the patch was merged after all in `0.8.0`.

I'm going to fix one new error and it's good to go.
